### ISSUE TITLE
Expand Boilerplate to handle importing css & images?

### ISSUE THREAD

**Issue by joshburgess** [cloverfield-tools/universal-react-boilerplate#95]
I recently started a new project using this boilerplate as a base, and I'm having a rather hard time trying to figure out how to go about handling important components that import css & images (ex: logos or css pulled in by icon fonts that are used in an app's main navigation components, etc.).

I think it would be nice to show a working example with a little more complexity than just the blank page render. How are you all handling this? So far, I've been checking the process.env.BROWSER variable and doing conditional requires() and have managed to serve the page, but haven't yet figured out how to properly pick it all up on the client-side.

**Comment by ericelliott** [comment]
Can you share your existing code?

**Comment by joshburgess** [comment]
Unfortunately, I can't due to the nature of the product (single page app for an enterpise company), but I could talk with you on Gitter/Skype/Google Hangouts/Twitter if you have some free time. I was just trying to figure out how to use this stack & server-side rendering while also handling rendering CSS/images. Have been reading about it a lot the past few days, and it seems it's still handled in a variety of different ways by different people.

**Comment by ericelliott** [comment]
> I was just trying to figure out how to use this stack & server-side rendering while also handling importing & rendering CSS/images in the main components that I'd want to serve up (like the navigation on the main screen of the app).

I personally don't use CSS or image imports using JavaScript modules. Is that what you're talking about?

**Comment by joshburgess** [comment]
Yeah, I'm importing them like you would JS modules, and webpack is able to handle this during the build steps just fine for client-side rendering, but it breaks when with server-side rendering. Ideally I'd like to have these sorts of things included on that first load, since it involves logos/icons/css used by the navigational components seen in the initial screen of the app.

You just link straight to css/images/etc. in standard HTML from the static assets folder served up by Express?

**Comment by ericelliott** [comment]
> but it breaks with server-side rendering

This is to be expected, because it relies on browser APIs to work. If you want full control over these things, you need to roll your own. AFAIK, Webpack does not currently support universal module loading for CSS & images.

**Comment by yanivkalfa** [comment]
@joshburgess && @ericelliott , webpack does support them if your build your server like you would client - with target:'node', however i am having hard time with that.

**Comment by yanivkalfa** [comment]
i made a lot of progress with that the past weekend/ i have the build/production working fine. i am now working on the development mode. once i have something concrete i will try to fork/offer it here.

**Comment by ericelliott** [comment]
Thanks. I'm not sure we want to make the decision to use CSS modules part of this boilerplate, though. There are other ways to handle CSS, and I'm not convinced importing it as modules is the best approach for everybody.

**Comment by yanivkalfa** [comment]
@ericelliott I am not talking about css modules, i am talking about simply requiring css anywhere you like, and extract-text-webpack-plugin picking it up into another file. the way it is now, when you render on the server you have to either ignore css/fonts/images etc. or something like that. it limit the use of external plugins no ?

maybe i am unfamiliar with another way mind elaborating on that ?

**Comment by ericelliott** [comment]
>  I am not talking about css modules, i am talking about simply requiring css anywhere you like, and extract-text-webpack-plugin picking it up into another file.

That's what css modules are. https://github.com/webpack/css-loader

**Comment by ericelliott** [comment]
There are lots of other options. See [this comparison](https://github.com/MicheleBertoli/css-in-js) for an overview. Common alternatives are specifying css in JavaScript, and even ignoring CSS completely from the JS and just using CSS with HTML as it was originally intended.

**Comment by yanivkalfa** [comment]
Well not quite modules are: https://github.com/css-modules/css-modules.  you have to specifically enable them with this loader, require them into an object and later use the object.propertyAsClassName 

and the css is all message up and ugly.

I am talking about completely normal css you will in your react component do something like:

```
import styless from './aStyle.css';

<component className={"aClassName"}/> 

Vs

<component className={styless.aClassName}/> 
```

**Comment by ericelliott** [comment]
Yeah, `css-modules/css-modules` is just *one implementation* of loading css as modules. There are many. ;)
